storage/azure/src/integration-test/java/io/aiven/kafka/tieredstorage/storage/azure/AzureBlobStorageSocks5Test.java

```diff
@@ -0,0 +1,175 @@
+
+    static BlobServiceClient blobServiceClient;
+
+    static String azuriteContainerNetworkAlias = AZURITE_SERVER.getNetworkAliases().get(0);
+
+    // This endpoint points to Azurite with the Docker-internal hostname and port.
+    // The host running these tests cannot reach it directly, only via a proxy.
+    static String internalConnectionString;
+
+    protected String azureContainerName;
+
+    @BeforeAll
+    static void setUpClass() {
+        final String connectionStringWithMappedPort = connectionString(AZURITE_SERVER, BLOB_STORAGE_PORT);
+        blobServiceClient = new BlobServiceClientBuilder()
+            .connectionString(connectionStringWithMappedPort)
+            .buildClient();
+
+        internalConnectionString = "DefaultEndpointsProtocol=http;"
+            + "AccountName=devstoreaccount1;"
+            + "AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;"
+            + "BlobEndpoint=" + String.format("http://%s:%d/devstoreaccount1", azuriteContainerNetworkAlias, BLOB_STORAGE_PORT) + ";";
+    }
+
+    @BeforeEach
+    void setUp(final TestInfo testInfo) {
+        azureContainerName = testInfo.getDisplayName()
+            .toLowerCase()
+            .replace("(", "")
+            .replace(")", "");
+        while (azureContainerName.length() < 3) {
+            azureContainerName += azureContainerName;
+        }
+        blobServiceClient.createBlobContainer(azureContainerName);
+    }
+
+    @Test
+    void worksWithAuthenticatedProxy() throws StorageBackendException {
+        // Ensure we can access the storage through the proxy with authentication.
+        final var proxy = PROXY_AUTHENTICATED;
+        final AzureBlobStorage storage = new AzureBlobStorage();
+        storage.configure(Map.of(
+            "azure.container.name", azureContainerName,
+            "azure.connection.string", internalConnectionString,
+            "proxy.host", proxy.getHost(),
+            "proxy.port", proxy.getMappedPort(SOCKS5_PORT),
+            "proxy.username", SOCKS5_USER,
+            "proxy.password", SOCKS5_PASSWORD
+        ));
+
+        storage.upload(new ByteArrayInputStream(new byte[] {0, 1, 2, 3}), OBJECT_KEY);
+        storage.fetch(OBJECT_KEY);
+    }
+
+    @Test
+    void worksWithUnauthenticatedProxy() throws StorageBackendException {
+        // Ensure we can access the storage through the proxy without authentication.
+        final var proxy = PROXY_UNAUTHENTICATED;
+        final AzureBlobStorage storage = new AzureBlobStorage();
+        storage.configure(Map.of(
+            "azure.container.name", azureContainerName,
+            "azure.connection.string", internalConnectionString,
+            "proxy.host", proxy.getHost(),
+            "proxy.port", proxy.getMappedPort(SOCKS5_PORT)
+        ));
+
+        storage.upload(new ByteArrayInputStream(new byte[] {0, 1, 2, 3}), OBJECT_KEY);
+        storage.fetch(OBJECT_KEY);
```

Comment:
Should not we check the content?

Reply:
I'm not sure, sounds redundant

Reply:
I added the content check to not use `assertThatNoException().isThrownBy(...)`.